src/db/user_session.rs

```diff
@@ -34,19 +35,21 @@ impl DbConn {
                 schema::users::avatar_url.eq(excluded(schema::users::avatar_url)),
             ))
             .get_result(self.inner())
-            .map_err(|_| DatabaseError::InsertUserFailed(user.github_login.clone()))?;
+            .map_err(|err| DatabaseError::InsertUserFailed(user.github_login.clone(), err))?;
 
         let new_session = models::NewSession {
             user_id: saved_user.id,
-            expires_at: SystemTime::now() + Duration::from_secs(u64::from(expires_in)),
+            expires_at: DateTime::<Utc>::from(
+                SystemTime::now() + Duration::from_secs(u64::from(expires_in)),
+            ),
```

Comment:
can we not add duration to a utc time itself? i.e. get utc timenow then add duration to it; rather than gettting local time and doing this